CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+# Siglens Contributor Guide
+
+* [New Contributor Guide](#contributing-guide)
+  * [Ways to Contribute](#ways-to-contribute)
+  * [Find an Issue](#find-an-issue)
+  * [Ask for Help](#ask-for-help)
+  * [Pull Request Lifecycle](#pull-request-lifecycle)
+  * [Development Environment Setup](#development-environment-setup)
+  * [Pull Request Checklist](#pull-request-checklist)
+
+Hello there! We are glad that you want to contribute to our project! 💖
+
+As you get started, you are in the best position to give us feedback on areas of
+our project that we need help with including:
+
+* Problems found during setting up a new developer environment
+* Gaps in our documentation
+
+If anything doesn't make sense, or doesn't work when you run it, please open a
+bug report and let us know!
+
+## Ways to Contribute
+
+
+We welcome many different types of contributions including:
+
+* New features
+* Bug fixes
+* Documentation
+* Issue Triage
+
+Not everything happens through a GitHub pull request. Please join our Slack: [SigLens Community](https://www.siglens.com/slack) and let's discuss how we can work together. 
+
+
+## Find an Issue
+
+We have good first issues for new contributors and help wanted issues suitable
+for any contributor. [good first issue](https://github.com/siglens/siglens/labels/good%20first%20issue) has extra information to help you make your first contribution. 
+[help wanted](https://github.com/siglens/siglens/labels/help%20wanted) are issues suitable for someone who isn't a core maintainer and is good to move onto after your first pull request.
+
+Sometimes there won’t be any issues with these labels. That’s ok! There is
+likely still something for you to work on. If you want to contribute but you
+don’t know where to start or can't find a suitable issue, you can 
+join our slack [SigLens Community](https://www.siglens.com/slack)
+
+Once you see an issue that you'd like to work on, please post a comment saying
+that you want to work on it. Something like `"I want to work on this"` is fine.
+
+## Ask for Help
+
+
+The best way to reach us with a question when contributing is to ask on:
+
+* The original github issue
+* Our Slack channel : [SigLens Community](https://www.siglens.com/slack)
+
+## Pull Request Lifecycle
+
+Once you have found the issue to be fixed or feature to be added, you can comment on the issue and put the approach you want to follow to solve the issue. Once we agree upon the approach, you can open a PR.
+
+1. Fork Siglens repo and clone it on your local machine.
+2. Make sure local tests work.
+3. Commit your changes to your fork.
```

Comment:
Are you saying to run tests immediately after cloning as a sanity check, or do they add code to fix the issue in step 1.5?

Reply:
good catch, fixed it. Added a TBD about makefiles, when they are ready, we will change this steps to run the make cmd.